src/API/TimesheetController.php

```diff
@@ -527,4 +530,91 @@ public function stopAction($id)
 
         return $this->viewHandler->handle($view);
     }
+
+    /**
+     * Restarts a previously stopped timesheet record for the current user
+     *
+     * @SWG\Response(
+     *      response=200,
+     *      description="Restarts a timesheet record for the same customer, project, activity combination. The current user will be the owner of the new record. Kimai tries to stop running records, which is expected to fail depending on the configured rules. Data will be copied from the original record if requested.",
+     *      @SWG\Schema(ref="#/definitions/TimesheetEntity")
+     * )
+     * @SWG\Parameter(
+     *      name="id",
+     *      in="path",
+     *      type="integer",
+     *      description="Timesheet record ID to restart",
+     *      required=true,
+     * )
+     *
+     * @Rest\RequestParam(name="copy", requirements="all|tags|description", strict=true, nullable=true, description="Whether description and tags are copied to the new entry. Allowed values: all, tags, description (default: nothing is copied)")
+     *
+     * @Security("is_granted('start_own_timesheet') or is_granted('start_other_timesheet')")
+     *
+     * @param int $id
+     * @return Response
+     * @throws \App\Repository\RepositoryException
+     * @throws \Doctrine\ORM\ORMException
+     * @throws \Doctrine\ORM\OptimisticLockException
+     */
+    public function restartAction($id, ParamFetcherInterface $paramFetcher, ValidatorInterface $validator)
+    {
+        /** @var Timesheet $timesheet */
+        $timesheet = $this->repository->find($id);
+        /** @var User $user */
+        $user = $this->getUser();
+
+        if (null === $timesheet) {
+            throw new NotFoundException();
+        }
+
+        if ($timesheet->getUser()->getId() !== $user->getId() && !$this->isGranted('start_other_timesheet')) {
```

Comment:
unneccessary, is covered in the voter next call `if (!$this->isGranted('start', $timesheet)) {`